### DISCUSSION TITLE
MCMC: Sharp right skewed gaussian distribution of corner plot

### DISCUSSION THREAD

**Discussion by ZareefJafar** [AstroBridge/BDLensing#207] [Q&A]
For DESIJ1554+0443 system the corner plot of `R_sersic_source_light0` shows Sharp right skewed gaussian distribution and `e1_source_light0` shows left skewed gaussian distribution. There are some light leaks in the center as it can be seen in the `source light plot` . Not sure if these two are correlated. 

Should I change the lower and upper limit of `R_sersic` and  `e1`  for the source galaxy's light model? 

It will also be helpful if anyone can tell me about the `beta` parameter in the our model and how it is affecting the output of our model. [notebook](https://github.com/ZareefJafar/BDLensing/blob/bc19a5214006596e949f764fa048d77eff8482b2/lens_systems/DESIJ1554%2B0443/lens_model_DESIJ1554%2B0443.ipynb)

Total sampling iterations: 550000
N_step: 2750
N_walkers: 200
N_params: 22
reduced X^2 : 0.8800

{'kwargs_lens': 
[{'theta_E': 1.5091842328739984, 'gamma': 1.8901014239497564, 'e1': -0.041099116379646354, 'e2': -0.0956621081003019, 'center_x': -0.29394975901765913, 'center_y': -0.11433919329114163}, {'gamma1': 0.10353333874128645, 'gamma2': 0.03911900330449401, 'ra_0': 0, 'dec_0': 0}], 

'kwargs_source': 
[{'amp': 13.000988741441493, 'R_sersic': 0.19945103584868337, 'n_sersic': 1.0, 'e1': -0.4138424403687778, 'e2': -0.3948397889509597, 'center_x': -0.47458198488536235, 'center_y': -0.0883675283016383}, {'amp': array([ 84.96398627, -30.59160518, -19.34648437,  91.70154188,
        -3.96773323,  -4.56338646, -43.17135789,  12.25814388,
        12.10668521,  -7.56215783,  33.46001008, -18.81097522,
       -17.97431499,  15.16432014,  16.6892171 ,  -2.90940515,
        -7.65698778,  24.19051357, -12.4772358 ,  -7.26126556,
         1.30966518]), 'n_max': 5, 'beta': 0.14522999080218182, 'center_x': -0.47458198488536235, 'center_y': -0.0883675283016383}],

 'kwargs_lens_light': 
[{'amp': -61.68322423955752, 'R_sersic': 1.671186256462338, 'n_sersic': 4.0, 'e1': 0.04978152617559737, 'e2': 0.13602166202537538, 'center_x': -0.33034457405127343, 'center_y': -0.09603204602418842}, {'amp': 119.77363741163205, 'R_sersic': 1.3578637501676494, 'n_sersic': 4.0, 'e1': 0.03020504153176204, 'e2': 0.07777787869722155, 'center_x': -0.33034457405127343, 'center_y': -0.09603204602418842}], 'kwargs_ps': [], 'kwargs_special': {}, 'kwargs_extinction': []}

Thanks!!!

![output2](https://github.com/ajshajib/BDLensing/assets/49613484/b576c8bf-7b27-4dd1-8a26-efc8054af10b)

![plots](https://github.com/ajshajib/BDLensing/assets/49613484/fd1c9c24-5b27-4095-815d-81c1902725ee)

![plots2](https://github.com/ajshajib/BDLensing/assets/49613484/b2edff7a-b098-47a2-a819-e163e06b9f0f)

**Comment by Rajit13** [comment]
The first suggestion when you upload it to GitHub make the file less cluttered with commented-out codes. Vai same problem with you. Your model has `n_params=22` but you used way more walkers. Which is actually making the calculation heavy.

**Reply by ZareefJafar** [reply]
sorry for the cluttered code. I usually do the cleaning before a pull request. Will keep that in mind.

Yes the number of walkers makes the computation heavy but is it related to the R_sersic_source_light0 and e1_source_light0 convergence as well as light leak problem? I think the problem is somewhere else. I am not sure.

**Comment by rafee-adnan** [comment]
Corner plot does not carry much information about the run, as it summarizes the end part of the plot, which also can be seen from the trace plot. So kindly share the trace plot here. From the corner plot, it seems to me `R_sersic_source_light0` reached its upper limit and wants to go beyond.

**Reply by ZareefJafar** [reply]
Yes,  `R_sersic_source_light0` reached wants to go beyond upper limit. Here is the trace plot. I will soon implement the changes that we discussed in the meeting. `burnin = 2500` and `thin = 20 `for corner plot. 

![output](https://github.com/ajshajib/BDLensing/assets/49613484/25eac317-035e-4f62-ad9a-a0c2dc8a0ecf)
